src/panels/lovelace/cards/hui-sensor-card.ts

```diff
@@ -0,0 +1,375 @@
+    cardConfig.detail =
+      cardConfig.detail === 1 || cardConfig.detail === 2
+        ? cardConfig.detail
+        : 1;
+
+    this._config = cardConfig;
+  }
+
+  public getCardSize(): number {
+    return 3;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+    return html`
+      ${this.renderStyle()}
+      <ha-card @click="${this._handleClick}">
+        ${
+          !stateObj
+            ? html`
+                <div class="not-found">
+                  Entity not available: ${this._config.entity}
+                </div>
+              `
+            : html`
+                <div class="flex">
+                  <div class="icon">
+                    <ha-icon
+                      .icon="${this._config.icon || stateIcon(stateObj)}"
+                    ></ha-icon>
+                  </div>
+                  <div class="header">
+                    <span class="name"
+                      >${this._config.name || computeStateName(stateObj)}</span
+                    >
+                  </div>
+                </div>
+                <div class="flex info">
+                  <span id="value">${stateObj.state}</span>
+                  <span id="measurement"
+                    >${
+                      this._config.unit ||
+                        stateObj.attributes.unit_of_measurement
+                    }</span
+                  >
+                </div>
+                <div class="graph">
+                  <div>
+                    ${
+                      this._config.graph !== "none" &&
+                      stateObj.attributes.unit_of_measurement
+                        ? until(
+                            this._getHistory().then((history) => {
+                              return svg`
+                        <svg width="100%" height="100%" viewBox="0 0 500 100">
+                          <path
+                            d="${history}"
+                            fill="none"
+                            stroke="var(--accent-color)"
+                            stroke-width="5"
+                            stroke-linecap="round"
+                            stroke-linejoin="round"
+                          />
+                        </svg>
+                      `;
+                            }),
+                            html``
+                          )
+                        : ""
+                    }
+                  </div>
+                </div>
+              `
+        }
+      </ha-card>
+    `;
+  }
+
+  private _handleClick(): void {
+    fireEvent(this, "hass-more-info", { entityId: this._config!.entity });
+  }
+
+  private _coordinates(
+    history: any,
+    hours: number,
+    width: number,
+    detail: number
+  ): number[][] {
+    history = history.filter((item) => !Number.isNaN(Number(item.state)));
```

Comment:
We should first convert all items to numbers, then check for nan. Now you're converting to number again on line 147 and 148!